packages/domain/lib/usecase/get_movie_anticipated_response_usecase.dart

```diff
@@ -0,0 +1,40 @@
+import '../utils/const.dart';
+import '../model/movie_anticipated_response.dart';
+import '../repository/movie_repository.dart';
+import 'usecase.dart';
+
+class GetMovieAnticipatedResponseUseCase
+    implements UseCase<Future<List<MovieAnticipatedResponse>>> {
+  final NetworkRepository _repository;
+
+  GetMovieAnticipatedResponseUseCase(this._repository);
+
+  @override
+  Future<List<MovieAnticipatedResponse>> call() async {
+    final List<MovieAnticipatedResponse> jsonMovies = [];
+    final response =
+        await _repository.getMovieData(apiPath: C.apiAnticipatedPath);
+    if (int.parse(response.headers['x-pagination-page-count'][0]) >= 5) {
+      const int itemCount = 50;
+      await _addToJsonMovies(itemCount, jsonMovies);
+    } else {
+      final itemCount =
+          int.parse(response.headers['x-pagination-item-count'][0]);
+      await _addToJsonMovies(itemCount, jsonMovies);
```

Comment:
the same as higher
example:

1) There are 50 films
2) u take first 20
3) after u fetched first 20 u check how many films are at the server (there are still 50)
4) u take 50 (with 20 u already fetched), while u should take 30 next